### PR TITLE
Add new component <Stack />

### PR DESCRIPTION
**Pull request by cdedreuille**
**Overview**
As discuss this is a first attempt to create a new layout component called Stack to evenly space children elements. It also simplifies the alignement using left, center or right. Perhaps later it might be interesting to look into adding a divider between each children that could be themed in v6. 

**Screenshots (if applicable)**
Screenshot 2020-10-11 at 00 58 32

**Documentation**
- [x] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [x] Added / modified Storybook stories

### CODE REVIEW

File: src/stackComponent/src/Stack.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const Stack = memo(
+  forwardRef(function Stack(props, ref) {
+    const { className, children, space, align, ...restProps } = props
+
+    let justifyItems = align
+    if (align === 'left') justifyItems = 'start'
+    if (align === 'right') justifyItems = 'end'
+
+    return (
+      <Pane
```
Comment (colinlohner) [inline src/stackComponent/src/Stack.js:15]: Could we actually swap this `<Pane />` out for a `<Box />` from `ui-box`? We are trying to reduce inter-component dependencies

**Review by colinlohner — CHANGES_REQUESTED** [review]

**Comment by colinlohner** [comment]
👍 Thank you for making this PR!

File: src/stackComponent/src/Stack.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React, { memo, forwardRef } from 'react'
```
Comment (mshwery) [inline src/stackComponent/src/Stack.js:1]: Can we move this into the `/layers` directory alongside Pane/Card?

File: src/stackComponent/src/Stack.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types'
+import { spacing, dimensions, position, layout } from 'ui-box'
+import { Pane } from '../../layers'
+
+const Stack = memo(
```
Comment (mshwery) [inline src/stackComponent/src/Stack.js:6]: We _could_ introduce a breaking change and export this as `Stack`. I think we'd have to rename our existing zindex `Stack` component as something else – maybe `Layer`?

**Pull request closed without merging**